src/evict/evict_page.c

```diff
@@ -482,17 +482,18 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref,
 		WT_ASSERT(session,
 		    !__wt_page_is_modified(page) ||
 		    LF_ISSET(WT_SKIP_UPDATE_RESTORE));
-	} else {
-		/*
-		 * If the page was ever modified, make sure all of the updates
-		 * on the page are old enough they can be discarded from cache.
-		 */
-		if (!exclusive && mod != NULL &&
-		    !__wt_txn_visible_all(session, mod->rec_max_txn))
-			return (EBUSY);
 	}
 
 	/*
+	 * If the page was ever modified, make sure all of the updates
+	 * on the page are old enough they can be discarded from cache.
+	 */
+	if (!exclusive && mod != NULL &&
+	    !__wt_txn_visible_all(session, mod->rec_max_txn) &&
+	    !LF_ISSET(WT_SKIP_UPDATE_RESTORE))
+		return (EBUSY);
+
+	/*
```

Comment:
@michaelcahill: is it a worthwhile for debugging to assert this case even if exclusive is set?